rest-service/manager_rest/cluster_status_manager.py

```diff
@@ -428,14 +449,19 @@ def append_service_result(pm, res):
                 'status') == NodeServiceStatus.ACTIVE:
             service_results[dm]['status'] = res['status']
 
-        # Upate the instances part
+        # Update the instances part
         if pm in service_results[dm].get('extra_info', {}):
             # res' process manager is already present in corresponding result
             if 'instances' not in service_results[dm]['extra_info'][pm]:
                 service_results[dm]['extra_info'][pm]['instances'] = []
-            # ... append an instance to the list of instances
-            service_results[dm]['extra_info'][pm]['instances'].append(
-                res['extra_info'][pm].get('instances'))
+            # keep only one instance per host
+            for res_instance in res['extra_info'][pm].get('instances'):
+                if res_instance.get('host') not in [
+                        instance.get('host') for instance in
+                        service_results[dm]['extra_info'][pm]['instances']]:
```

Comment:
This is madness, but rather a necessary one. We want to be backwards compatible, thus we should keep this extremely deeply nested structure.